lib/puppet/application/apply.rb

```diff
@@ -185,17 +185,25 @@ def main
       facts.name = Puppet[:node_name_value]
     end
 
-    configured_environment = Puppet.lookup(:current_environment)
+    # Find our Node
+    unless node = Puppet::Node.indirection.find(Puppet[:node_name_value])
+      raise "Could not find node #{Puppet[:node_name_value]}"
```

Comment:
When can this indirection actually return `nil`? Could this be a clearer error that captures the information a user might need to fix it?

Also, should this raise Puppet::Error instead of a string?